client/client-multi/client-c/src/kaa/platform-impl/common/sha.c

```diff
@@ -24,7 +24,7 @@
 
 kaa_error_t ext_calculate_sha_hash(const char *data, size_t data_size, kaa_digest digest)
 {
-    if (!digest || !data || !data_size) {
+    if (!digest || (!data && data_size)) {
```

Comment:
I think it's better to add one more condition `(data && !data_size)` :

```
    if (!digest || (data_size && !data) || (data && !data_size)) {
        return KAA_ERR_BADPARAM;
    }
```

In this case the function's behavior won't change comparing to original function.  Passing the 0 size to function `ext_calculate_sha_hash(valid_data, 0, hash)` shouldn't result in error `KAA_BAD_PARAM`. Also we won't need to modify unit tests.